samples-js/functions/smsPhoneVerification.js

```diff
@@ -0,0 +1,76 @@
+const { output } = require("@azure/functions");
+const df = require("durable-functions");
+const { DateTime } = require("luxon");
+
+const sendSmsChallengeActivityName = "sendSmsChallenge";
+
+if (process.env.TwilioPhoneNumber && process.env.TwilioAccountSid && process.env.TwilioAuthToken) {
+    df.app.orchestration("smsPhoneVerification", function* (context) {
+        const phoneNumber = context.df.getInput();
+        if (!phoneNumber) {
+            throw new Error("A phone number input is required.");
+        }
+
+        const challengeCode = yield context.df.callActivity(
+            sendSmsChallengeActivityName,
+            phoneNumber
+        );
+
+        // The user has 90 seconds to respond with the code they received in the SMS message.
+        const expiration = DateTime.fromJSDate(context.df.currentUtcDateTime).plus({ hours: 90 });
+        const timeoutTask = context.df.createTimer(expiration.toJSDate());
+
+        let authorized = false;
+        for (let i = 0; i <= 3; i++) {
+            const challengeResponseTask = context.df.waitForExternalEvent("SmsChallengeResponse");
+
+            const winner = yield context.df.Task.any([challengeResponseTask, timeoutTask]);
+
+            if (winner === challengeResponseTask) {
+                // We got back a response! Compare it to the challenge code.
+                if (challengeResponseTask.result === challengeCode) {
+                    authorized = true;
+                    break;
+                }
+            } else {
+                // Timeout expired
+                break;
+            }
```

Comment:
could you reverse this check so that you break if `(winner === timeoutTask)`? That would allow you to move that inner if statement one level out, which would help with readability

Reply:
Done, thanks :)